test_scripts/Polices/Policies_Security/Trigger_PTU_NO_Certificate/008_ATF_P_TC_navi_app_PTU_fail_ForceProtectedService_ON_StartService_encrypted_true.lua

```diff
@@ -0,0 +1,273 @@
+        end)
+    end)
+end
+
+function Test:TestStep_Audio_NACK()
+  self.mobileSession.correlationId = self.mobileSession.correlationId + 1
+
+  local msg = {
+    serviceType = 10,
+    frameInfo = 1,
+    frameType = 0,
+    rpcCorrelationId = self.mobileSession.correlationId,
+    encryption = true
+  }
+  self.mobileSession:Send(msg)
+
+  local startserviceEvent = Event()
+  startserviceEvent.matches =
+  function(_, data)
+    return ( data.frameType == 0 and data.serviceType == 10)
+  end
+
+  self.mobileSession:ExpectEvent(startserviceEvent, "Service 10: StartServiceNACK")
+  :ValidIf(function(_, data)
+      if data.frameInfo == 2 then
+        commonFunctions:printError("Service 10: StartServiceACK is received")
+        return false
+      elseif data.frameInfo == 3 then
+        print("Service 10: Audio NACK")
+        return true
+      else
+        commonFunctions:printError("Service 10: StartServiceACK/NACK is not received at all.")
+        return false
+      end
+    end)
+
+  commonTestCases:DelayedExp(10000)
+end
+
+function Test:TestStep_Video_NACK()
+  self.mobileSession.correlationId = self.mobileSession.correlationId + 1
+
+  local msg = {
+    serviceType = 11,
+    frameInfo = 1,
+    frameType = 0,
+    rpcCorrelationId = self.mobileSession.correlationId,
+    encryption = true
+  }
+  self.mobileSession:Send(msg)
+
+  local startserviceEvent = Event()
+  startserviceEvent.matches =
+  function(_, data)
+    return ( data.frameType == 0 and data.serviceType == 11)
+  end
+
+  self.mobileSession:ExpectEvent(startserviceEvent, "Service 11: StartServiceNACK")
+  :ValidIf(function(_, data)
+      if data.frameInfo == 2 then
+        commonFunctions:printError("Service 11: StartServiceACK is received")
+        return false
+      elseif data.frameInfo == 3 then
+        print("Service 11: Audio NACK")
+        return true
+      else
+        commonFunctions:printError("Service 11: StartServiceACK/NACK is not received at all.")
+        return false
+      end
+    end)
+
+  commonTestCases:DelayedExp(10000)
+end
+
+function Test:TestStep_RPC_NACK()
+  self.mobileSession.correlationId = self.mobileSession.correlationId + 1
+
+  local msg = {
+    serviceType = 7,
+    frameInfo = 0,
+    rpcType = 0,
+    rpcFunctionId = 48,
+    encryption = true,
+    rpcCorrelationId = self.mobileSession.correlationId,
+    payload = '{ "audioStreamingIndicator" : "PAUSE" }'
+  }
+  self.mobileSession:Send(msg)
+
+  local startserviceEvent = Event()
+  startserviceEvent.matches =
+  function(_, data)
+    return ( data.frameType == 0 and data.serviceType == 7)
+  end
+
+  self.mobileSession:ExpectEvent(startserviceEvent, "Service 7: StartServiceNACK")
+  :ValidIf(function(_, data)
+      if data.frameInfo == 2 then
+        commonFunctions:printError("Service 7: StartServiceACK is received")
+        return false
+      elseif data.frameInfo == 3 then
+        print("Service 7: Audio NACK")
```

Comment:
@istoimenova, Test is about 'RPC' service. Seems message has to be corrected. Or I propose to use something common for all services (like for NACK)